api/src/main/java/com/google/sps/SendNotifications.java

```diff
@@ -0,0 +1,150 @@
+package com.google.sps;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.firestore.CollectionReference;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import com.google.firebase.cloud.FirestoreClient;
+import com.google.api.services.gmail.model.Message;
+import com.google.api.services.gmail.Gmail;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Properties;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.internet.*;
+
+import org.apache.commons.codec.binary.Base64;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class SendNotifications {
+
+  public List<Email> gatherRecipients() throws IOException {
+    GoogleCredentials credentials = GoogleCredentials.getApplicationDefault();
+
+    System.out.println("Google Credentials created...");
+
+    String projectId = "phoebeliang-step";
+    FirebaseOptions options = new FirebaseOptions.Builder().setCredentials(credentials).setProjectId(projectId).build();
+    FirebaseApp.initializeApp(options);
+
+    System.out.println("App Initialized...");
+
+    Firestore db = FirestoreClient.getFirestore();
+
+    String gameID = "dT6U1wQKR1rGJcEncyMS";
+    List<Email> emails = new ArrayList<>();
+
+    DocumentReference game = db.collection("games").document(gameID);
+    CollectionReference players = db.collection("users");
+    try {
+      DocumentSnapshot gameDocSnap = game.get().get();
+      System.out.println("Is this the error?...");
+      List<String> playerNames = (List<String>) gameDocSnap.get("players");
+      System.out.println(playerNames);
+
+      for(String player: playerNames) {
+
+        DocumentSnapshot usersDocSnap = players.document(player).get().get();
+
+        //Retrieves player information
+        String playerEmail = usersDocSnap.getString("email");
+        String playerName = usersDocSnap.getString("username");
+
+        System.out.println(playerEmail);
+
+        //Adding player to Email object
+        emails.add(new Email(new User(playerEmail, playerName)));
+
+        //Adds Start game message to the email object
+        emails.get(emails.size()-1).playerTurn(gameID);
```

Comment:
Do we need any check here?

Reply:
there should always be one player in the game so i don't think it should be a problem